Subject: MIPS: Octeon: Mark SMP-IPI interrupt as IRQF_NO_THREAD
From: Venkat Subbiah <venkat.subbiah@caviumnetworks.com>
Date: Mon,  3 Oct 2011 13:31:10 -0700

From: Venkat Subbiah <venkat.subbiah@cavium.com>

This is to exclude it from force threading to allow RT patch set to work.
And while on this line
* Remove IRQF_DISABLED as as this flag is NOOP
* Add IRQF_PERCPU as this is a per cpu interrupt.


Signed-off-by: Venkat Subbiah <venkat.subbiah@cavium.com>
Acked-by: David Daney <david.daney@cavium.com>
Cc: linux-mips@linux-mips.org
Cc: ralf@linux-mips.org
Link: http://lkml.kernel.org/r/1317673870-10671-1-git-send-email-venkat.subbiah@caviumnetworks.com
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 arch/mips/cavium-octeon/smp.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: linux-3.2/arch/mips/cavium-octeon/smp.c
===================================================================
--- linux-3.2.orig/arch/mips/cavium-octeon/smp.c
+++ linux-3.2/arch/mips/cavium-octeon/smp.c
@@ -207,8 +207,9 @@ void octeon_prepare_cpus(unsigned int ma
 	 * the other bits alone.
 	 */
 	cvmx_write_csr(CVMX_CIU_MBOX_CLRX(cvmx_get_core_num()), 0xffff);
-	if (request_irq(OCTEON_IRQ_MBOX0, mailbox_interrupt, IRQF_DISABLED,
-			"SMP-IPI", mailbox_interrupt)) {
+	if (request_irq(OCTEON_IRQ_MBOX0, mailbox_interrupt,
+			IRQF_PERCPU | IRQF_NO_THREAD, "SMP-IPI",
+			mailbox_interrupt)) {
 		panic("Cannot request_irq(OCTEON_IRQ_MBOX0)\n");
 	}
 }
